### PR TITLE
fix comment misdirection during scaling loss

### PR DESCRIPTION
**Pull request by techkang**
# What does this PR do?
If the GA loss bug is not fixed, we should scale the loss properly. However, it is not solely for reporting purposes as described in the previous comment. https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L3764

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@SunMarc @muellerzr @ArthurZucker

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @muellerzr @SunMarc

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -3752,7 +3752,7 @@ def training_step(
             with amp.scale_loss(loss, self.optimizer) as scaled_loss:
                 scaled_loss.backward()
         else:
-            # Finally we need to normalize the loss for reporting
+            # Finally we need to normalize the loss if GA loss bug is not fixed during compute loss
             if not self.model_accepts_loss_kwargs and self.compute_loss_func is None:
                 loss = loss / self.args.gradient_accumulation_steps
```
Comment (SunMarc) [inline src/transformers/trainer.py:3757]: I feel like this condition if not strong enough. Users can overwrite compute loss and not use `num_item_per_batch` at all when calculating the loss (e.g trl case as I explained [here](https://github.com/huggingface/transformers/pull/36426#discussion_r2014186216) ). The issue is that right now it won't scale the loss. 
Maybe we should create a variable to enable/disable the GA fix but in transformers, we set enable it since we are calculating the loss the right way.

**Review by SunMarc — COMMENTED** [review]
LGTM ! Left some thoughts regarding the condition below
